go.sum

```diff
@@ -328,6 +330,7 @@ github.com/oklog/run v1.0.0/go.mod h1:dlhp/R75TPv97u0XWUtDeV/lRKWPKSdTuV0TZvrmrQ
 github.com/onsi/ginkgo v1.6.0/go.mod h1:lLunBs/Ym6LB5Z9jYTR76FiuTmxDTDusOGeTQH+WWjE=
 github.com/onsi/ginkgo v1.8.0/go.mod h1:lLunBs/Ym6LB5Z9jYTR76FiuTmxDTDusOGeTQH+WWjE=
 github.com/onsi/gomega v1.5.0/go.mod h1:ex+gbHU/CVuBBDIJjb2X0qEXbFg53c61hWP/1CpauHY=
+github.com/opentracing/opentracing-go v1.1.0/go.mod h1:UkNAQd3GIcIGf0SeVgPpRdFStlNbqXla1AfSYxPUl2o=
```

Comment:
Unnecessary extra dependency? Do we care? This is the only extra dependency go-grpc-middleware brings.

Reply:
I find the PingService particularly useful for the tests. Otherwise, I would have to come up with a proto definition for a stream RPC and have the whole infra set here, including the go code generation out of the proto files. Or just ignore the test for the streaming part.

Reply:
To be clear: my preference is to use go-grpc-middleware, as it gives us the convenience and test coverage. I might even switch the configauth to use the auth parts of it in the future, as I've seen that their interceptors are very similar to what we have here.

Reply:
I understand that they are useful things, but also extra dependencies like OpenTracing and others. Can we work with that team to maybe split their module?

Reply:
Looks like there's an issue there that might be relevant: https://github.com/grpc-ecosystem/go-grpc-middleware/issues/238. It might take a while for this to happen though, given that there's not been any movement since Oct 2019.

To unblock this PR: should I remove the streaming tests, removing the dependency on go-grpc-middleware, or are we OK with this dependency, assuming it's going to be temporary?

In any case, this dependency doesn't seem to leak all the way down to the binary, just to the module resolution:
```
$ strings bin/cmd-otelcol | grep github.com/grpc-ecosystem/go-grpc-middleware
,github.com/grpc-ecosystem/go-grpc-middleware
dep	github.com/grpc-ecosystem/go-grpc-middleware	v1.3.0	h1:+9834+KizmvFV7pXQGSXQTsaWhq2GjuNUt0aUU0YBYw=
github.com/grpc-ecosystem/go-grpc-middleware.(*WrappedServerStream).Context
type..eq.github.com/grpc-ecosystem/go-grpc-middleware.WrappedServerStream
github.com/grpc-ecosystem/go-grpc-middleware.(*WrappedServerStream).RecvMsg
github.com/grpc-ecosystem/go-grpc-middleware.(*WrappedServerStream).SendHeader
github.com/grpc-ecosystem/go-grpc-middleware.(*WrappedServerStream).SendMsg
github.com/grpc-ecosystem/go-grpc-middleware.(*WrappedServerStream).SetHeader
github.com/grpc-ecosystem/go-grpc-middleware.(*WrappedServerStream).SetTrailer
github.com/grpc-ecosystem/go-grpc-middleware.WrappedServerStream.RecvMsg
github.com/grpc-ecosystem/go-grpc-middleware.WrappedServerStream.SendHeader
github.com/grpc-ecosystem/go-grpc-middleware.WrappedServerStream.SendMsg
github.com/grpc-ecosystem/go-grpc-middleware.WrappedServerStream.SetHeader
github.com/grpc-ecosystem/go-grpc-middleware.WrappedServerStream.SetTrailer
github.com/grpc-ecosystem/go-grpc-middleware.WrapServerStream
github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/wrappers.go

$ strings bin/cmd-otelcol | grep opentr
$
```

Reply:
This is now fixed, the opentracing dependency isn't there anymore.